src/centralserver/admin-service/src/test/java/org/niis/xroad/centralserver/restapi/openapi/SystemApiControllerTest.java

```diff
@@ -26,26 +26,51 @@
  */
 package org.niis.xroad.centralserver.restapi.openapi;
 
+import ee.ria.xroad.signer.protocol.dto.TokenInfo;
+
+import org.junit.Before;
 import org.junit.Test;
-import org.junit.jupiter.api.BeforeEach;
+import org.niis.xroad.centralserver.openapi.model.ServerAddressUpdateBody;
 import org.niis.xroad.centralserver.openapi.model.SystemStatus;
 import org.niis.xroad.centralserver.openapi.model.TokenInitStatus;
 import org.niis.xroad.centralserver.openapi.model.Version;
+import org.niis.xroad.centralserver.restapi.service.SystemParameterService;
+import org.niis.xroad.centralserver.restapi.util.TokenTestUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.ResponseEntity;
 import org.springframework.security.test.context.support.WithMockUser;
 
+import javax.validation.ConstraintViolationException;
+
+import static ee.ria.xroad.commonui.SignerProxy.SSL_TOKEN_ID;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.niis.xroad.centralserver.restapi.service.SystemParameterService.CENTRAL_SERVER_ADDRESS;
+import static org.niis.xroad.centralserver.restapi.service.SystemParameterService.INSTANCE_IDENTIFIER;
 
 public class SystemApiControllerTest extends AbstractApiControllerTestContext {
 
     @Autowired
     SystemApiController systemApiController;
 
-    @BeforeEach
-    void setUp() {
+    private TokenInfo testSWToken;
+
+    @MockBean
+    private SystemParameterService systemParameterService;
```

Comment:
`@MockBean` annotation causes the testing context to become dirty -> new context gets created. 

This became a big problem with the proxy-ui-api when every test was launching a new test context and running unit tests took near 15 minutes (a lot of tests). 

So in this case it would be good that `AbstractApiControllerTestContext` would do the mocking.

Also as a side note, `AbstractFacadeMockingTestContext` class mocks `SignerProxyService` at the moment (the file name implies that it only mocks the facade layer). That would probably fit better in the `AbstractApiControllerTestContext` class.

Reply:
Fixed in: 
f58e954